exercise/exercise_models.py

```diff
@@ -713,7 +724,21 @@ def load(self, request, students, url_name="exercise"):
         if not students:
             return ExercisePage(self)
 
-        url = self.service_url or self.lti_service.url
+        replace_url = None
+        if self.service_url:
+            parse_result = urllib.parse.urlparse(self.service_url)
+            if parse_result.netloc:
+                # Deprecated: absolute URL stored in the database. Drop the domain now.
+                # Typically, LTI exercises do not need to specify a service URL since
+                # they usually use the launch URL of the LTI service. If a custom
+                # service URL is needed, it must be in the same domain as the
+                # LTI service. Previously, LTI exercises stored service URLs in the
+                # absolute form, but now only the URL path is stored.
+                replace_url = drop_domain_from_url(parse_result=parse_result)
```

Comment:
`replace_url = parse_result._replace(scheme='', netloc='').geturl()`

Reply:
`_replace` method is not part of the public API...

Reply:
Hmm.. True. Result object is documented to be subclass of `tuple` not `namedtuple` and  `_replace` is part of public api for `namedtuple`.

I personally see that `_replace` is the cleanest way, but if we go with just tuples then `type(r)('', '', *r[2:]).geturl()` or `urlunsplit(('', '', *r[2:]))`.

ps. all instances of urlparse can be replaced with urlsplit in the code.

Reply:
I suppose we can assume that `_replace` is part of the public API, since the documentation in `urllib.parse` says that the return value is a tuple that has named attributes. Normal tuples do not have such attributes, but named tuples do, and the implementation of the library uses them.